db/versions/0017.yml

```diff
@@ -0,0 +1,242 @@
+version: 17
+migrationScript: 0017-migration.sql
+downgradeScript: 0017-downgrade.sql
+methods:
+  wmworker_pool_errors_entities_load:
+    description: See taskcluster-lib-entities
+    mode: read
+    serviceName: worker_manager
+    args: partition_key text, row_key text
+    returns: table (partition_key_out text, row_key_out text, value jsonb, version integer, etag uuid)
+    body: |-
+      begin
+        return query
+        select
+          wmworker_pool_errors_entities_load.partition_key,
+          wmworker_pool_errors_entities_load.row_key,
+          entity_buf_encode(
+            jsonb_build_object(
+              'PartitionKey', encode_string_key(error_id),
+              'RowKey', encode_string_key(error_id),
+              'errorId', error_id,
+              'workerPoolId', worker_pool_id,
+              'reported', reported,
+              'kind', kind,
+              'title', title,
+              'description', description),
+            'extra', extra::text) as value,
+          1 as version,
+          worker_pool_errors.etag as etag
+        from worker_pool_errors
+        where
+          worker_pool_errors.error_id = decode_string_key(wmworker_pool_errors_entities_load.partition_key);
+      end
+  wmworker_pool_errors_entities_create:
+    serviceName: worker_manager
+    description: See taskcluster-lib-entities
+    mode: write
+    args: pk text, rk text, properties jsonb, overwrite boolean, version integer
+    returns: uuid
+    body: |-
+      declare
+        new_row worker_pool_errors%ROWTYPE;
+      begin
+        select
+          (properties ->> 'errorId')::text as error_id,
+          (properties ->> 'workerPoolId')::text as worker_pool_id,
+          (properties ->> 'reported')::timestamptz as reported,
+          (properties ->> 'kind')::text as kind,
+          (properties ->> 'title')::text as title,
+          (properties ->> 'description')::text as description,
+          entity_buf_decode(properties, 'extra')::jsonb as extra,
+          public.gen_random_uuid() as etag
+        into new_row;
+        if overwrite then
+          raise exception 'overwrite not implemented';
+        else
+          execute 'insert into worker_pool_errors select $1.*' using new_row;
+        end if;
+        return new_row.etag;
+      end
+  wmworker_pool_errors_entities_remove:
+    serviceName: worker_manager
+    description: See taskcluster-lib-entities
+    mode: write
+    args: partition_key text, row_key text
+    returns: table (etag uuid)
+    body: |-
+      begin
+        return query delete from worker_pool_errors
+        where
+          worker_pool_errors.error_id = decode_string_key(wmworker_pool_errors_entities_remove.partition_key)
```

Comment:
I think the `where` clause should also specify the partition key `worker_pool_id`. Same thing with the other where clauses in this file.